src/v/v8_engine/data_policy_table.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "model/metadata.h"
+#include "v8_engine/data_policy.h"
+
+#include <absl/container/node_hash_map.h>
+
+#pragma once
+
+namespace v8_engine {
+
+class data_policy_table {
+public:
+    using container_type
+      = absl::node_hash_map<model::topic_namespace, v8_engine::data_policy>;
+
+    bool insert(model::topic_namespace, v8_engine::data_policy);
+
+    bool erase(model::topic_namespace);
+
+    std::optional<v8_engine::data_policy>
+    get_data_policy(const model::topic_namespace& topic) const;
+
+    const container_type& get_current_state() const { return _dps; }
```

Comment:
can we replace `get_current_state` with the access methods you performing manually on the container?